src/components/singleArticle.js

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticleByID } from "../api";
+
+export default function ArticleByID() {
+  const [article, setArticle] = useState([]);
+  const [isLoading, setLoading] = useState(true);
+  const { article_id } = useParams();
+
+  useEffect(() => {
+    fetchArticleByID(parseInt(article_id))
+      .then(({ articles }) => {
+        setLoading(true);
```

Comment:
I suggest moving this setLoading before your `fetchArticleByID` given how Promises work!

Reply:
Thanks Karla will implement changes now :)